File: src/services/get_available_crous.rs
use crate::{
  api::available_crous::{make_request, parse_response},
  models::Crous,
};
use reqwest::Error;

pub async fn get_available_crous() -> Result<Vec<Crous>, Error> {
  let response = make_request().await?;
  Ok(parse_response(response))
}

File: src/services/getAvailableCrous.ts
import type { Crous } from "~/models/Crous";
import { buildRequest, parseResponse } from "~/api/availableCrous";
import { type Fetcher, defaultFetcher } from "@literate.ink/utilities";

export async function getAvailableCrous(fetcher: Fetcher = defaultFetcher): Promise<Array<Crous>> {
  const response = await fetcher(buildRequest());
  return parseResponse(response);
}
